src/System.Private.ServiceModel/src/Internals/System/Runtime/TaskHelpers.cs

```diff
@@ -179,16 +179,121 @@ public static void ToApmEnd(this IAsyncResult iar)
             }
         }
 
-        // Task.GetAwaiter().GetResult() calls an internal variant of Wait() which doesn't wrap exceptions in
-        // an AggregateException.
-        public static void WaitForCompletion(this Task task)
+        public static void RunSyncWithParams<T1>(this Func<T1, Task> function, T1 param1)
         {
-            task.GetAwaiter().GetResult();
+            var state = Tuple.Create(function, param1);
+            var syncTask = new Task<Task>(obj => {
+                var tuple = obj as Tuple<Func<T1, Task>, T1>;
+                var func = tuple.Item1;
+                var p1 = tuple.Item2;
+                return func(p1);
+            }, state, CancellationToken.None);
```

Comment:
Question only: I don't have a good sense of what it means to have CancellationToken.None and still use a timeout parameter for most of these.  Does this defeat timeouts?